src/log/log.c

```diff
@@ -2163,8 +2255,10 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 		 * Read the minimum allocation size a record could be.
 		 */
 		WT_ASSERT(session, buf->memsize >= allocsize);
+		need_salvage = true;
```

Comment:
Setting it to true/false around this `wt_read` call is subtle. I assume it is to change what happens in the event of an error. It needs a comment to make clear what it's doing and why it's okay.

Reply:
Added a comment, the code looks slightly different after refactoring.